go/worker/sentry/grpc/worker.go

```diff
@@ -153,67 +135,15 @@ func (g *Worker) authFunction() auth.AuthenticationFunction {
 	}
 }
 
-func (g *Worker) updatePolicies(p policyAPI.ServicePolicies) {
-	g.logger.Debug("updating policies",
-		"policy", p,
-	)
-
-	g.Lock()
-	defer g.Unlock()
-
-	g.grpcPolicyCheckers[p.Service] = policy.NewDynamicRuntimePolicyChecker(p.Service, nil)
-	for namespace, policy := range p.AccessPolicies {
-		g.grpcPolicyCheckers[p.Service].SetAccessPolicy(policy, namespace)
-	}
-}
-
 func (g *Worker) worker() {
 	defer close(g.quitCh)
 	defer (g.cancelCtx)()
 
-	if g.upstreamConn == nil {
-		dialUpstream := func() error {
-			_, err := g.upstreamDialer(g.ctx)
-			if err != nil {
-				return err
-			}
-			return nil
-		}
-
-		sched := backoff.WithMaxRetries(backoff.NewConstantBackOff(1*time.Second), 60)
-		err := backoff.Retry(dialUpstream, backoff.WithContext(sched, g.ctx))
-		if err != nil {
-			g.logger.Error("unable to dial upstream node",
-				"err", err,
-			)
-			return
-		}
-	}
-
-	// Initialize policy watcher.
-	g.policyWatcher = policyAPI.NewPolicyWatcherClient(g.conn)
-	ch, sub, err := g.policyWatcher.WatchPolicies(g.ctx)
```

Comment:
Since there is no longer need for a policy watcher client (the `PolicyWatcherClient` interface), it and all uses should be removed. However the `PolicyWatcher` interface still seems useful to centralize policy updates (see my other comment).